soc/silabs/silabs_siwx917/siwg917/nwp_init.c

```diff
@@ -25,7 +25,11 @@ static int siwx917_nwp_init(void)
 		.band = SL_SI91X_WIFI_BAND_2_4GHZ,
 		.region_code = DEFAULT_REGION,
 		.boot_config = {
+#ifdef CONFIG_WIFI_SIWX917_AP_MODE
+			.oper_mode = SL_SI91X_ACCESS_POINT_MODE,
+#else
 			.oper_mode = SL_SI91X_CLIENT_MODE,
+#endif
```

Comment:
I believe this does not make sense. The use probably want AP for provisioning (= retrieve credential of the WiFi network). Then he will want to start the station mode.

This should be a runtime option.